subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractModuleComponentResolveMetadata.java

```diff
@@ -135,6 +137,19 @@ public ModuleSource getSource() {
     }
 
     @Override
+    public List<? extends ConfigurationMetadata> getConsumableConfigurationsHavingAttributes() {
+        if (consumableConfigurations == null) {
```

Comment:
Both.

Don't worry about the cost, at least until it shows up in the profiler, as these instances are cached in memory, and only created when it's likely we're just about to select configurations from it.